Main.hs

```diff
@@ -362,16 +400,16 @@ compileFun' :: Definition -> LocalDecls -> TCM [Hs.Decl ()]
 compileFun' (Defn {..}) locals = do
   let n = qnameName defName
       x = hsName $ prettyShow n
-      go = foldM $ \(ds, ms) -> compileClause ds defName x >=> return . fmap (ms `snoc`)
+      go = foldM $ \(ds, ms) -> compileClause ds x >=> return . fmap (ms `snoc`)
   setCurrentRange (nameBindingSite n) $ do
     ty <- compileType (unEl defType)
     cs <- snd <$> go (locals, []) funClauses
     return [Hs.TypeSig () [x] ty, Hs.FunBind () cs]
   where
     Function{..} = theDef
 
-compileClause :: LocalDecls -> QName -> Hs.Name () -> Clause -> TCM (LocalDecls, Hs.Match ())
```

Comment:
I removed this as it doesn't seem to be used. The next arg Hs.Name () is just a rendering of the `QName` which is used instead.